### PR TITLE
Fix to #74223 , leaderelection.go affirming creation of new log and l…

### LINKED ISSUE #74223: Leaderelect.go doesnt affirm it created a new lock for a new leader
Looking at controller/scheduler logs, You get a somewhat counterintuitive read out in cases where a cluster goes down eventually: 

```
{"log":"E0218 03:22:53.751888       1 leaderelection.go:224] error retrieving resource lock kube-system/kube-controller-manager: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member http://127.0.0.1:2379 has no leader\n","stream":"stderr","time":"2019-02-18T03:22:53.752513991Z"}
{"log":"E0218 09:34:58.980155       1 leaderelection.go:224] error retrieving resource lock kube-system/kube-controller-manager: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member http://127.0.0.1:2379 has no leader\n","stream":"stderr","time":"2019-02-18T09:34:58.980472645Z"}
``` 

In this case, we had a healthy controller manager for 6 hours or so, and then something went wrong.  at *9:34*.  However, since all we see is errors by default, we cant easily see that in a crashlog.  This PR just makes it so we affirm that newly created locks get reported.  It also levels note about trying to acquire a lock function so that its logs are on the same level as logs in the election flow .

### PR DESCRIPTION
**Pull request by jayunit100**
…eveled log of acquire trial 

**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:

Ideally, the leader elect should show that its creating a new lock, so that logs aren't 100% negative when looking at a cluster that is healthy / became corrupt.

**Which issue(s) this PR fixes**:

Fixes #74223  

**Special notes for your reviewer**:

This ones a Quickie :) 

**Does this PR introduce a user-facing change?**:
```release-note
Improved leaderelection lock logging for DR/failure forensics scenarios
```

### CODE REVIEW

**Comment by jayunit100** [comment]
flake on 1`[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: tmpfs] [Testpattern: Pre-provisioned PV (default fs)] volumes should be mountable 1m32s` ?

**Comment by jayunit100** [comment]
Just ran into this again today - could we move this PR forward ? Makes life so much easier with multimaster clusters.

File: staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (le *LeaderElector) acquire(ctx context.Context) bool {
 		succeeded = le.tryAcquireOrRenew()
 		le.maybeReportTransition()
 		if !succeeded {
-			klog.V(4).Infof("failed to acquire lease %v", desc)
+			klog.Infof("failed to acquire lease %v", desc)
```
Comment (jpbetz) [inline staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: Won't `failed to acquire lease` be logged repeatedly from the non-leaders for expected cases like "lock is held by %v and has not yet expired", with no way to turn it off, if we remove `V(4)`? Are there error cases in `tryAcquireOrRenew` do we need more logging for that don't already log to error?

File: staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go [hunk 2]
```diff
@@ -301,17 +301,20 @@ func (le *LeaderElector) tryAcquireOrRenew() bool {
...
 			return false
 		}
+		// Try to create the lock
 		if err = le.config.Lock.Create(leaderElectionRecord); err != nil {
 			klog.Errorf("error initially creating leader election record: %v", err)
 			return false
 		}
 		le.observedRecord = leaderElectionRecord
 		le.observedTime = le.clock.Now()
+		klog.Info("Created a new lock, and acquired it successfully at time: %v", le.observedTime)
```
Comment (jpbetz) [inline staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: "Created a new leader election record" ?  "Lock" is a bit vague.

"successfully acquired lease" or "successfully renewed lease" messages are already logged after this, are they sufficient or can they be improved to log what is needed?

**Review by jpbetz — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go (hunk 1 above)
Comment (jayunit100) [reply staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: This is a good point, I havent checked on this PR in a while.  
I wouldnt mind a periodic Infof that rolls up that the last 10 acquires for lease failed, that way its not spammy.

File: staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go (hunk 2 above)
Comment (jayunit100) [reply staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: fixed

File: staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go (hunk 1 above)
Comment (jayunit100) [reply staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: Hi @jpbetz   .. Ive updated this to only report once every 10 times which i think means that the message will generally pop up every minute or so, which is i think appropriate.    Let me know what you think ?

Comment (jayunit100) [reply staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go]: (obsolete now, removed this change)

**Pull request closed without merging**